myia/type/unity.py

```diff
@@ -0,0 +1,173 @@
+from typing import (Dict as DictT, Tuple as TupleT, Iterable, Union,
```

Comment:
I'd rename the file to `unify` or `unification`.

Reply:
I like the name `unity`, so I'll keep it for now.  @bartvm can be the tiebreaker.